jta-1_2-in-wildfly/pom.xml

```diff
@@ -38,21 +38,20 @@
 
     <properties>
         <project.build.sourceEncoding>UTF-8</project.build.sourceEncoding>
-        <maven.compiler.target>1.7</maven.compiler.target>
-        <maven.compiler.source>1.7</maven.compiler.source>
         <version.org.jboss.spec.javax.transaction>1.0.1.Final</version.org.jboss.spec.javax.transaction>
         <version.org.jboss.spec.javax.ejb>1.0.0.Final</version.org.jboss.spec.javax.ejb>
         <version.org.jboss.spec.javax.jms>1.0.0.Final</version.org.jboss.spec.javax.jms>
         <version.hibernate.javax.persistence>1.0.0.Final</version.hibernate.javax.persistence>
         <version.javax.enterprise>1.1</version.javax.enterprise>
         <version.junit>4.13.1</version.junit>
-        <version.org.wildfly.arquillian>2.1.1.Final</version.org.wildfly.arquillian>
+        <version.org.wildfly.arquillian>3.0.1.Final</version.org.wildfly.arquillian>
         <version.org.jboss.arquillian>1.1.13.Final</version.org.jboss.arquillian>
         <ipv6.server.jvm.args>-Djboss.bind.address=[::1] -Djboss.bind.address.management=[::1] -Djboss.bind.address.unsecure=[::1] -Djava.net.preferIPv4Stack=false -Djava.net.preferIPv6Addresses=true</ipv6.server.jvm.args>
         <jvm.args.memory>-Xms64m -Xmx1024m -XX:MaxMetaspaceSize=512m</jvm.args.memory>
         <jvm.args.debug></jvm.args.debug>
         <jvm.args.other>-server</jvm.args.other>
         <version.surefire.plugin>2.19.1</version.surefire.plugin>
+        <modular.jdk.args>--add-opens=java.base/java.util=ALL-UNNAMED</modular.jdk.args>
```

Comment:
Shouldn't this go in a profile?

Reply:
Thank you for your question @jmfinelli. Actually it should. But the only way to make it work is to add it to the parent profile, otherwise other profiles might not see other profile's properties. Unfortunately there is not way to establish an order of profiles in the same pom. WDYT if I create a parent pom and I put this quickstart in a subdirectory? So I could insert the profile in the 'just created' pom.

Reply:
I am inferring a bit here what is being asked for and what the response means but @marcosgopen, are you certain that if you don't have a jdk17 profile that defined the property modular.jdk.args then the arqIPv6 can't reliably pick it up? Something like that is done to get https://github.com/jbosstm/narayana/blob/master/txframework/pom.xml#L205 for `jdk17` and I am not sure but are failures being reported for JDK 17 builds not having --add-opens set?

Reply:
What about if you leave the property `modular.jdk.args` empty here and then you define a `jdk17` profile (that gets active only when JDK >= 17) where you fill `modular.jdk.args` with `--add-opens`?

Reply:
Basically
```
<properties>
    <!-- populated using the jdk17 profile -->
    <modular.jdk.args></modular.jdk.args>
</properties>
<profile> 
    <id>jdk17</id>
        <activation>
          <jdk>[17,)</jdk>
        </activation>
        <properties>
           <modular.jdk.args>--add-opens=java.base/java.util=ALL-UNNAMED</modular.jdk.args>
        </properties>
    </profile>
</profiles>
```

Reply:
@jmfinelli that is what I tried and I expected it to work. But when I run "mvn clean install" arq profile is executed and the modular.jdk.args property is empty (jdk17 profile does not seem to be executed). That is why it fails and I ended up using the property.

Reply:
I found out that if we define an activation property (i.e. !ipv6) in the arq profile the behaviour is different: maven wait all the profiles to define variables (i.e. ${modular.jdk.args}) and then execute arq profile correctly (quite odd behaviour) that means the modular.jdk.args variable is correctly defined. I think this might be a solution.

Reply:
I did not dive very deep into this issue but I just tested your branch. If I define a `jdk17` profile as I suggested above, the quickstart does not activate the `arq` profile (which should be actived by default :confused: ). I agree, strange behaviour. Anyway, if `-Parq` is added to `mvn clean test` (or `./build.sh` etc.), everything works as it should. If you meant that this (i.e. `P-arq`) might be a workaround, I agree :-) OTOH, it would be nice to understand why `arq` does not get activated 

BTW, I also tried to delete `<modular.jdk.args></modular.jdk.args>` from the `properties` section and (as expected) the quickstart works fine. So I think we can only use the `jdk17` profile to define `modular.jdk.args`

Reply:
I think you are right and the problem is the activeByDefault configuration for arq profile. If you agree I would replace it with an activation property which works best.

Reply:
I think that activating an Arquillian profile by default is not very nice. As this configuration is giving us problem now (very strangly :confused: ), it could be a good occasion to change the behaviour, WDYT? I like how Narayana handles Arquillian profiles. Narayana defines a variable (ARQ) to run a specific Arquillian profile. This allows Narayana to run different profile based on the behaviour the user wants to test (for example, managed, embedded, IP6). We could develop something like that in our quickstarts, WDYT? (In case you like this idea, we could track this improvement with another JIRA)

Reply:
If you take off activeByDefault you can just activate it on the command line with `-Parq` (assuming the profile is called `arq`), you can activate multiple `-Parq,profile2,...`

I recently made a change to have a property `arqProfileActivated` so that I could activate arq and jdk17 profile only in the condition that it was JDK 17 too and otherwise only arq would activate. Hopefully see this in the changes to `scripts/hudson/narayana.sh` and `ArjunaJTA/jta/pom.xml` in https://github.com/jbosstm/narayana/commit/61aa3be08363f7dfe60c9897c0b99b005ab12520

I am clarifying here a bit because typically Narayana had different profiles rather than a variable to control what Arquillian profile is ran but I might be misfollowing and this might not be helpful and in which case please excuse the noise.

Reply:
Narayana uses a variable in its [script](https://github.com/jbosstm/narayana/blob/f6b9472c6f20f72cb49b647cbccfc0e46985fd58/scripts/hudson/narayana.sh#L84) (`narayana.sh`). It seems that [this](https://github.com/jbosstm/narayana/commit/61aa3be08363f7dfe60c9897c0b99b005ab12520#diff-234fb7d762fdc80a3f29e8d8f4e0669ad47101d550eb3724420e91f64ee044deL538) instruction should have been using `-P$ARQ_PROF` :confused: .

Reply:
Thanks @jmfinelli - do you think that we should provide more than one arq profile in that jta one?

I think even with more than one Arquillian profile we can make something work. We can ask that the profile is specified in the `arqProfileActivated` property rather than `true`, e.g. in the `arq` profile:
```      <activation>
        <activeByDefault>false</activeByDefault>
        <property>
          <name>arqProfileActivated</name>
          <value>arq</value>
        </property>
      </activation>```

Then in the narayana.sh we can have:
`./build.sh -f ArjunaJTA/jta/pom.xml -fae -B -DarqProfileActivated=$ARQ_PROF $CODE_COVERAGE_ARGS "$@" test`

Reply:
Thanks @tomjenkinson and @jmfinelli for your response. I can see an easy way to use the activation property and does not need to set an additional profile as maven argument. Here is the example: https://github.com/jbosstm/quickstart/pull/403/commits/bb4ba9856d10a6b833bb33b75dc3e8e75e06cfc4#diff-a78913a2bd252946c65e5393c9d22a7f0e360157ef53e33b22ca36ec2b3610b6R160-R162

Reply:
I took the idea from other quickstart: https://github.com/jbosstm/quickstart/blob/master/XTS/wsat-jta-multi_service/pom.xml#L225

Reply:
> do you think that we should provide more than one arq profile in that jta one

Thanks for highlighting this :-). Yeah, maybe could be an improvement and it would be in line with other module in ArjunaJTA (i.e. `cdi`).

About the activation of the `arq` profile, I would prefer to stick with `-P$ARQ_PROF` in the build script instead of defining a property (which is in line with other modules in Narayana)

Reply:
Re: the arquillian profiles in Narayana under ArjunaJTA I have created https://issues.redhat.com/browse/JBTM-3683

Reply:
I think we should err on having the profiles activated by users. That being said I think for my case I wanted to activate the jdk17 when Maven was running with JDK17 and if the Arquillian tests should be ran. Only in those two situations (specifically it is not necessary to activate the JDK17 profile even when using JDK 17 unless also using the `arq` profile).

I don't have a strong opinion that we shouldn't do that kind of thing I mention there where it's not straight forward, but I do agree that hidden triggers should be avoided. I think customizing the build based on the environment (rather than properties) seems OK and the alternative seems to be someone building with JDK17 needs to also tell our build that they are using jdk17 and that could be seen as a bit inefficient or something?

Reply:
> I think we should err on having the profiles activated by users

Could you please share the reason why activating maven profiles in quickstarts should error?

Reply:
Sorry, I was meaning I was supporting generally having the profiles by users (but not exclusively). I have edited my previous comment.